pkg/ebpf/probes/probes.go

```diff
@@ -100,7 +96,8 @@ func Init(module *bpf.Module, netEnabled bool) (Probes, error) {
 		SecurityFileMProtect:       &traceProbe{eventName: "security_file_mprotect", probeType: kprobe, programName: "trace_security_file_mprotect"},
 		CommitCreds:                &traceProbe{eventName: "commit_creds", probeType: kprobe, programName: "trace_commit_creds"},
 		SwitchTaskNS:               &traceProbe{eventName: "switch_task_namespaces", probeType: kprobe, programName: "trace_switch_task_namespaces"},
-		ARMKprobe:                  &traceProbe{eventName: "arm_kprobe", probeType: kprobe, programName: "trace_arm_kprobe"},
+		EnableKprobe:               &traceProbe{eventName: "enable_kprobe", probeType: kprobe, programName: "trace_enable_kprobe"},
+		EnableKprobeRet:            &traceProbe{eventName: "enable_kprobe", probeType: kretprobe, programName: "trace_ret_enable_kprobe"},
```

Comment:
`arm_kprobe` is called not only by `enable_kprobe`, but also from `register_kprobe`, and at least one more place. Are you sure this is enough to use it?

Reply:
I did a rush fix and did not read the arm/enable logic, failed to see other callers. I tried other probe points, but they failed (and I thought it was because of the kprobe being in a critical path at that time, thus the reason why I changed the kprobe).

Nevertheless, reading a bit more now, I see that all arches have `arch_arm_kprobe()` function and symbol exists for both: amd64 and arm64.

I'll move the probe from `enable_kprobe()` to `arch_arm_kprobe()` instead. The enable/disable logic is not the same as arming and disarming the probe, which is the logic that replaces the instructions by the kprobe jumps de-facto.

Reply:
I'm still sending the event in kretprobe (instead of kprobe) so I can check if the arming was successful or not. Should we only send the event when successful OR send the retcode together with the event @yanivagman and @roikol ?

Reply:
With that said, I believe that if we check:

1. `enable_kprobe()`
2. `register_kprobe()`
3. `register_aggr_kprobe()` (might not be needed as it is called by `register_kprobe()`

kretprobes return value, we might have the result from the kprobes arm/disarm logic.

Reply:
I'll add the other 2 probes tomorrow and create an inline function called by all the related probes. Question: since I'm generating the event from the kretprobe now, and I do have the ret code, should we generate the event for failed attempts as well ? Or, should we include the ret code in the event to tell if enablement was successful ?

Reply:
No need for the 2 probes.. `register_kprobe` and `enable_kprobe` give duplicate events (same execution path). `register_kprobe` is better because it also might be able to call `register_aggr_kprobes` (in case of multiple `kprobes`) and it covers all cases that could be calling `arm_kprobe` (which can't be directly hooked to as explained before).